### PR TITLE
[intesis] Add configurable polling interval

### PR DESCRIPTION
**Pull request by hmerk**
Signed-off-by: Hans-Jörg Merk [github@hmerk.de](mailto:github@hmerk.de)

### CODE REVIEW

File: bundles/org.openhab.binding.intesis/README.md [hunk 1]
```diff
@@ -19,11 +19,12 @@ Intesis devices do not support auto discovery.
...
-|-----------|---------------------|----------------------------------------------------------------|
-| ipAddress | Both                | IP-Address of the device                                       |
-| password  | IntesisHome         | Password to login to the local webserver of IntesisHome device |
-| port      | IntesisBox          | TCP port to connect to IntesisBox device, defaults to 3310     |
+| Parameter        | Valid for ThingType | Description                                                    |
+|------------------|---------------------|----------------------------------------------------------------|
+| ipAddress        | Both                | IP-Address of the device                                       |
+| password         | IntesisHome         | Password to login to the local webserver of IntesisHome device |
+| port             | IntesisBox          | TCP port to connect to IntesisBox device, defaults to 3310     |
+| pollingIntervall | Both                | Interval to retrieve updates from the connected devices        |
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/README.md:27]:
```suggestion
| pollingInterval  | Both                | Interval to retrieve updates from the connected devices        |
```

File: bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/config/IntesisBoxConfiguration.java [hunk 2]
```diff
@@ -23,4 +23,5 @@
 public class IntesisBoxConfiguration {
     public String ipAddress = "";
     public int port;
+    public int pollingIntervall;
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/config/IntesisBoxConfiguration.java:26]:
```suggestion
    public int pollingInterval = 45;
```

File: bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/config/IntesisHomeConfiguration.java [hunk 3]
```diff
@@ -23,4 +23,5 @@
 public class IntesisHomeConfiguration {
     public String ipAddress = "";
     public String password = "";
+    public int pollingIntervall;
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/config/IntesisHomeConfiguration.java:26]:
```suggestion
    public int pollingInterval = 30;
```

File: bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisBoxHandler.java [hunk 4]
```diff
@@ -107,14 +106,13 @@ public void initialize() {
...
                     intesisLocalApi.sendAlive();
-
                 } catch (IOException e) {
                     updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
                     return;
                 }
                 updateStatus(ThingStatus.ONLINE);
             });
-            pollingTask = scheduler.scheduleWithFixedDelay(this::polling, 3, 45, TimeUnit.SECONDS);
+            pollingTask = scheduler.scheduleWithFixedDelay(this::polling, 3, config.pollingIntervall, TimeUnit.SECONDS);
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisBoxHandler.java:115]:
```suggestion
            pollingTask = scheduler.scheduleWithFixedDelay(this::polling, 3, config.pollingInterval, TimeUnit.SECONDS);
```

File: bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java [hunk 5]
```diff
@@ -466,7 +466,7 @@ private void handleDataPointsResponse(Response response) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
         }
         logger.trace("Start Refresh Job");
-        refreshJob = scheduler.scheduleWithFixedDelay(this::getAllUidValues, 0, INTESIS_REFRESH_INTERVAL_SEC,
+        refreshJob = scheduler.scheduleWithFixedDelay(this::getAllUidValues, 0, config.pollingIntervall,
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java:469]:
```suggestion
        refreshJob = scheduler.scheduleWithFixedDelay(this::getAllUidValues, 0, config.pollingInterval,
```

File: bundles/org.openhab.binding.intesis/src/main/resources/OH-INF/thing/thing-types.xml [hunk 6]
```diff
@@ -22,6 +22,15 @@
 				<description>@text/thing-type.config.intesisHome.password.description</description>
 				<context>password</context>
 			</parameter>
+			<parameter name="pollingInterval" type="decimal" min="30" unit="s">
+				<label>Polling Interval</label>
+				<description>
+					Defines the time in seconds for openHAB to pull the
+					state of the connected devices. The minimum is 60
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/src/main/resources/OH-INF/thing/thing-types.xml:29]:
```suggestion
					state of the connected devices. The minimum is 30
```

File: bundles/org.openhab.binding.intesis/src/main/resources/OH-INF/thing/thing-types.xml [hunk 7]
```diff
@@ -22,6 +22,15 @@
 				<description>@text/thing-type.config.intesisHome.password.description</description>
 				<context>password</context>
 			</parameter>
+			<parameter name="pollingInterval" type="decimal" min="30" unit="s">
+				<label>Polling Interval</label>
+				<description>
+					Defines the time in seconds for openHAB to pull the
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/src/main/resources/OH-INF/thing/thing-types.xml:28]:
```suggestion
					Defines the time in seconds to pull the
```

File: bundles/org.openhab.binding.intesis/src/main/resources/OH-INF/thing/thing-types.xml [hunk 8]
```diff
@@ -46,6 +55,15 @@
 				<description>The TCP port to the IntesisBox.</description>
 				<default>3310</default>
 			</parameter>
+			<parameter name="pollingInterval" type="decimal" min="45" unit="s">
+				<label>Polling Interval</label>
+				<description>
+					Defines the time in seconds for openHAB to pull the
+					state of the connected devices. The minimum is 60
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/src/main/resources/OH-INF/thing/thing-types.xml:62]:
```suggestion
					state of the connected devices. The minimum is 45
```

File: bundles/org.openhab.binding.intesis/src/main/resources/OH-INF/thing/thing-types.xml [hunk 9]
```diff
@@ -46,6 +55,15 @@
 				<description>The TCP port to the IntesisBox.</description>
 				<default>3310</default>
 			</parameter>
+			<parameter name="pollingInterval" type="decimal" min="45" unit="s">
+				<label>Polling Interval</label>
+				<description>
+					Defines the time in seconds for openHAB to pull the
```
Comment (jlaur) [inline bundles/org.openhab.binding.intesis/src/main/resources/OH-INF/thing/thing-types.xml:61]:
```suggestion
					Defines the time in seconds to pull the
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks. I have added some comments, but also have a concern that it doesn't work at all because of a typo: `thing-tyles.xml` defines parameters as "pollingInterval", but it's deserialized as "pollingInterval**l**".

**Comment by hmerk** [comment]
@jlaur Anything else ? All your suggestions have been committed.

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by jlaur** [comment]
> Anything else ? All your suggestions have been committed.

Sorry, I had missed your commits. Looks good now. 👍 Can you confirm it has been tested now, because I think it could not have worked with the initial commit?

**Comment by hmerk** [comment]
Yes, tested and working.

**Pull request merged**